sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -825,8 +1030,9 @@ def _set_dtype(self, X, sample_weight, centers_init):
             # The other cases would not trigger any additional memory copies.
             X = X.astype(compute_dtype)
 
-        centers_init_dtype = centers_init.dtype
-        if centers_init.dtype != compute_dtype:
+        if centers_init is not None and (
+            (centers_init_dtype := centers_init.dtype) != compute_dtype
```

Comment:
Nice use of the walrus operator! :)
Surprisingly enough, that's the first time I get to see it in a review!

Reply:
I'm a big fan

You don't see it a lot probably because a lot of software still support python 3.7 =)

Also many people dislike because of readability